examples/plot_unified_validation_scheme.py

```diff
@@ -0,0 +1,344 @@
+    torch.backends.cudnn.benchmark = True
+seed = 20200220
+set_random_seeds(seed=seed, cuda=cuda)
+
+n_classes = 4
+# Extract number of chans and time steps from dataset
+n_chans = train_set[0][0].shape[0]
+input_window_samples = train_set[0][0].shape[1]
+
+model = ShallowFBCSPNet(
+    n_chans,
+    n_classes,
+    input_window_samples=input_window_samples,
+    final_conv_length='auto',
+)
+
+# Send model to GPU
+if cuda:
+    model.cuda()
+
+
+######################################################################
+# Option 1: Simple Train-Test Split
+# ---------------------------------
+#
+
+
+######################################################################
+# This is the easiest training scheme to use as the dataset is only
+# splitted in two distinct sets (``train_set`` and ``test_set``).
+# As this method uses no separate validation split it should only be
+# used for the final evaluation of the (previously!) found
+# Hyperparameter configuration.
+#
+# .. warning::
+#    If you make any use of the ``test_set`` during training
+#    (i.e. by using EarlyStopping) there will be data leakage
+#    which will make the reported generalization capability/decoding
+#    performance of your model less credible.
+#
+
+from skorch.callbacks import LRScheduler
+
+from braindecode import EEGClassifier
+
+lr = 0.0625 * 0.01
+weight_decay = 0
+batch_size = 64
+n_epochs = 4
+
+clf = EEGClassifier(
+    model,
+    criterion=torch.nn.NLLLoss,
+    optimizer=torch.optim.AdamW,
+    train_split=None,
+    optimizer__lr=lr,
+    optimizer__weight_decay=weight_decay,
+    batch_size=batch_size,
+    callbacks=[
+        "accuracy", ("lr_scheduler", LRScheduler('CosineAnnealingLR', T_max=n_epochs - 1)),
+    ],
+    device=device,
+)
+# Model training for a specified number of epochs. `y` is None as it is already supplied
+# in the dataset.
+clf.fit(train_set, y=None, epochs=n_epochs)
+
+# score the Model after training
+y_test = test_set.get_metadata().target
+test_acc = clf.score(test_set, y=y_test)
+print(f"Test acc: {(test_acc * 100):.2f}%")
+
+
+######################################################################
+# Option 2: Fast Hyperparameter Search
+# ------------------------------------
+#
+
+######################################################################
+# Usually when developing a new Deep Learning model/method finding
+# the best (or at least a suitable) Hyperparameter configuration makes
+# up a substantial part of the developement process.
+# As stated above, it is not suitable to use the ```test_set``
+# for this Hyperparamter search. Therefore we need a third split, the
+# so called validation set which is a Subset of the ``train_set``.
+# This second option splits the original ``train_set`` only once
+# (instead of k times as in Option 3) to speed up the tuning process.
+# This method should only be preferred over Option 3 if either the
+# training duration is very long or the Hyperparamter search space is
+# very large.
```

Comment:
```suggestion
# Usually, when developing a new deep learning model/method finding 
# the best (or at least a suitable) hyperparameter configuration makes 
# up a substantial part of the development process. 
# 
#As stated above, using the ```test_set`` 
# for this hyperparameter search is unsuitable. Therefore we need a third split, 
# the so-called validation set, which is a Subset of the train_set. # This second scheme splits the original train_set only once 
# (instead of k times as in scheme 3) to speed up the tuning process. 
#
# This method should only be preferred over scheme 3 if the 
# training duration is very long, or the Hyperparameter search space # is very large.
```